backend/inspirehep/records/api/literature.py

```diff
@@ -89,6 +89,45 @@ def create(
             record.push_authors_phonetic_blocks_to_redis()
             return record
 
+    @classmethod
+    def get_es_linked_records_from_dict_field(cls, data, path):
+        """Return the generator of linked records from specified path for records in ES.
+
+        Args:
+            data (dict): data from which records should be extracted
+            path (str): the path of the linked records.
+        Yields:
+            InspireRecord: the linked records.
+        Examples:
+            >>> data = {
+                'references': [
+                    {
+                        'record': {
+                            '$ref': 'http://localhost/literature/1'
+                        }
+                    }
+                ]
+            }
+            >>>  records = LiteratureRecord.get_linked_records_from_field(
+                data, "references.record")
+        """
+        pids = cls._get_linked_pids_from_field(data, path)
+        if pids:
+            results = LiteratureSearch.get_records_by_pids(pids)
+            for result in results:
+                try:
+                    rec_data = json.loads(result._ui_display)
+                except AttributeError:
+                    LOGGER.exception(
+                        "Record do not have _ui_display field!", record=result.meta.id
```

Comment:
ops, `doesn't`.
Also, I think we have a lot of logs with recid reather than records uuid

Reply:
UUID here is the only thing which will be there always. recid might be missing from there if the record won't be indexed properly (and it's not indexed properly if _ui_display field is missing).